### PR TITLE
Maint: RepositoryConnector._decode_response

### PR DESCRIPTION
**Pull request by AhmetNSimsek**

### CODE REVIEW

File: siibra/retrieval/requests.py [hunk 1]
```diff
@@ -134,7 +148,15 @@ def find_suitiable_decoder(url: str):
         else:
             return None
 
-    def _set_decoder_func(self, func):
+    def _set_decoder_func(self, func: None):
```
Comment (xgui3783) [inline siibra/retrieval/requests.py:151]:
```suggestion
    def _set_decoder_func(self, func: Callable=None):
```

File: siibra/retrieval/requests.py [hunk 2]
```diff
@@ -119,7 +119,21 @@ def __init__(
...
+        url : str
+            The url or filename with extension.
+
+        Returns
+        -------
+        Callable or None
+        """
         urlpath = urllib.parse.urlsplit(url).path
         if urlpath.endswith(".gz"):
             dec = HttpRequest.find_suitiable_decoder(urlpath[:-3])
```
Comment (xgui3783) [inline siibra/retrieval/requests.py:139]: if `find_suitable_decode` returns `None` this code will break

File: siibra/retrieval/repositories.py [hunk 3]
```diff
@@ -43,15 +43,8 @@ def _build_url(self, folder: str, filename: str):
         pass
 
     def _decode_response(self, response, filename):
-        # see if we find a default encoder
-        suitable_decoders = [
-            dec for sfx, dec in DECODERS.items() if filename.endswith(sfx)
```
Comment (xgui3783) [inline siibra/retrieval/repositories.py:48]: is `DECODER` still used elsewhere? or the imports can also be removed?

**Review by xgui3783 — COMMENTED** [review]
it feels to me that `find_suitiable_decoder` can graduate and be on its own, if multiple classes are requiring it.

File: siibra/retrieval/repositories.py (hunk 3 above)
Comment (AhmetNSimsek) [reply siibra/retrieval/repositories.py:48]: There are a few places but I think I can remove them as they should be resolved automatically.

File: siibra/retrieval/requests.py (hunk 2 above)
Comment (AhmetNSimsek) [reply siibra/retrieval/requests.py:139]: Good point

**Pull request merged**